src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -132,9 +132,12 @@
    ca-settings :- ca/CaSettings]
   (let [response (ca/delete-certificate-request! ca-settings subject)
         outcomes->codes {:success 204 :not-found 404 :error 500}]
-    (-> (rr/response (:message response))
-        (rr/status ((response :outcome) outcomes->codes))
-        (rr/content-type "text/plain"))))
+    (if (= (response :outcome) :error) 
```

Comment:
Looks good, but we need a Content-Type (text/plain) for both the 404 and the 500 cases as the 404 also has a body (ex: `"No certificate request for not-found-agent at expected path /var/folders/dh/ccmxd9mn1kggz9stfrm04vyh0000gn/T/null1690930482830-3832794012/ca/requests/not-found-agent.pem"`)

Reply:
To expand, Content-Type should only be omitted if the body is empty (content-length of 0).